test/hotspot/jtreg/testlibrary_tests/template_framework/examples/TestTutorial.java

```diff
@@ -0,0 +1,539 @@
+package template_framework.examples;
+
+import java.util.Collections;
+import java.util.List;
+
+import compiler.lib.compile_framework.*;
+
+import compiler.lib.template_framework.Template;
+import compiler.lib.template_framework.Hook;
+import compiler.lib.template_framework.TemplateBinding;
+import compiler.lib.template_framework.Name;
+import static compiler.lib.template_framework.Template.body;
+import static compiler.lib.template_framework.Template.let;
+import static compiler.lib.template_framework.Template.$;
+import static compiler.lib.template_framework.Template.fuel;
+import static compiler.lib.template_framework.Template.addName;
+import static compiler.lib.template_framework.Template.sampleName;
+import static compiler.lib.template_framework.Template.weighNames;
+
+import compiler.lib.template_framework.library.Hooks;
+
+public class TestTutorial {
+
+    public static void main(String[] args) {
+        // Create a new CompileFramework instance.
+        CompileFramework comp = new CompileFramework();
+
+        // Add java source files.
+        comp.addJavaSourceCode("p.xyz.InnerTest1", generateWithListOfTokens());
+        comp.addJavaSourceCode("p.xyz.InnerTest2", generateWithTemplateArguments());
+        comp.addJavaSourceCode("p.xyz.InnerTest3", generateWithHashtagAndDollarReplacements());
+        comp.addJavaSourceCode("p.xyz.InnerTest4", generateWithCustomHooks());
+        comp.addJavaSourceCode("p.xyz.InnerTest5", generateWithLibraryHooks());
+        comp.addJavaSourceCode("p.xyz.InnerTest6", generateWithRecursionAndBindingsAndFuel());
+        comp.addJavaSourceCode("p.xyz.InnerTest7", generateWithNames());
+
+        // Compile the source files.
+        comp.compile();
+
+        // Object ret = p.xyz.InnterTest1.main();
+        comp.invoke("p.xyz.InnerTest1", "main", new Object[] {});
+        comp.invoke("p.xyz.InnerTest2", "main", new Object[] {});
+        comp.invoke("p.xyz.InnerTest3", "main", new Object[] {});
+        comp.invoke("p.xyz.InnerTest4", "main", new Object[] {});
+        comp.invoke("p.xyz.InnerTest5", "main", new Object[] {});
+        comp.invoke("p.xyz.InnerTest6", "main", new Object[] {});
+        comp.invoke("p.xyz.InnerTest7", "main", new Object[] {});
+    }
+
+    // This example shows the use of various Tokens.
+    public static String generateWithListOfTokens() {
+        // A Template is essencially a function / lambda that produces a
+        // token body, which is a list of Tokens that are concatenated.
+        var templateClass = Template.make(() -> body(
+            // The "body" method is filled by a sequence of "Tokens".
+            // This can be Strings and multi-line Strings, but also
+            // boxed primitives.
+            """
+            package p.xyz;
+
+            public class InnerTest1 {
+                public static void main() {
+                    System.out.println("Hello World!");
+            """,
+            "int a = ", Integer.valueOf(1), ";\n",
```

Comment:
Might be better to use `System.lineSeparator()` instead of `\n` to be platform independent.

Reply:
Hmm. You may be right. But then again, this also worked on Windows... what platform would it even fail on?

Reply:
I prefer multiline strings, but that does not always work. `\n` is just less of a disturbance.

Reply:
![image](https://github.com/user-attachments/assets/51e6af52-a8de-4d1f-93fd-f89b910f1310)

Boah, I feel like this really looks quite unreadable / distracting. It makes the lines super long.

Makes me look for other "hacky" solutions. Maybe some `#\n` string, to make a hashtag replacement that gets you the newline / lineSeparator?

![image](https://github.com/user-attachments/assets/226f7022-c799-4b04-867b-5233d8c9ee7c)
That looks a little better.

This is what I used to have:
![image](https://github.com/user-attachments/assets/3035aacf-ca09-4f5d-868e-d92a0adf2d04)

Reply:
Or should we just keep the `\n`, and wait until someone actually has an issue with it? Because on all of the platforms we run this on it works, even Windows where a lineSeparator() is supposed to be `\r\n`. And I'm a little hesitant to implement something heavy if we are not sure we really need it ;)

Reply:
@chhagedorn As discussed offline:
We figured out that text blocks also normalize newline to `\n`, see https://openjdk.org/jeps/378.
Since Templates will make heavy use of text blocks, it makes sense to just keep everything with `\n`.